node/src/components/fetcher/event.rs

```diff
@@ -10,23 +11,37 @@ use crate::{
     utils::Source,
 };
 
+#[derive(Clone, Debug, Error, PartialEq, Eq)]
+pub enum FetcherError<T, I>
+where
+    T: Item,
+    I: Debug + Eq,
+{
+    #[error("Could not fetch item with id {id:?} from peer {peer:?}")]
+    AbsentFromPeer { id: T::Id, peer: I },
```

Comment:
IMO the element not being present is not an error. Errors are _unexpected_/_invalid_ states while absent `AbsentFromPeer` is not one of them.

Reply:
Maybe not an error in this sense, but I think it makes sense in this context, as we reuse result with `Ok` meaning "we managed to fetch it" and `Err` meaning "we failed to fetch it". And this counts as a failure to fetch.

An alternative could be to define `FetchResult` as our own enum with variants like `Fetched` and `DidntFetch`, but I'm not sure it's worth it.

Reply:
And that's how it _was_ before (I already implemented that) and it's being changed in this PR (among other things).